includes/classes/productPulldown.php

```diff
@@ -0,0 +1,132 @@
+
+            $this->sort = ' ORDER BY pd.products_name';
+
+            $this->keyword_search_fields = [
+                'pd.products_name',
+                'p.products_model',
+                'pd.products_description',
+                'p.products_id',
+            ];
+        }
+
+        public function setSort($fieldnameArray)
+        {
+            if (empty($fieldnameArray)) {
+                return $this;
+            }
+
+            $first = true;
+            $this->sort = '';
+            foreach ($fieldnameArray as $fieldname) {
+                if (in_array($fieldname, $this->allowed_sort_array)) {
+                    $key = array_search($fieldname, $this->allowed_sort_array);
+                    $this->sort = ($first ? ' ORDER BY ' : ', ') . $this->db_table_prefix_array[$key] . $fieldname;
+                    $first = false;
+                }
+            }
+            return $this;
+        }
+
+        public function setCategory(int $category_id)
+        {
+            $this->categories_join = " LEFT JOIN " . TABLE_PRODUCTS_TO_CATEGORIES . " ptc ON (ptc.products_id = p.products_id)";
+            $this->condition .= " AND ptc.categories_id = " . (int)$category_id;
+            return $this;
+        }
+
+        public function showModel(bool $status)
+        {
+            $this->show_model = $status;
+            return $this;
+        }
+
+        public function showPrice(bool $status)
+        {
+            $this->show_price = $status;
+            return $this;
+        }
+
+        public function onlyActive(bool $status)
+        {
+        	$condition = " AND p.products_status = 1";
+        	$this->condition = str_replace($condition, '', $this->condition);
+        	if ($status) {
+		        $this->condition .= " AND p.products_status = 1";
+	        }
+            return $this;
+        }
+
+        protected function setSQL()
+        {
+            $this->sql = "SELECT DISTINCT pd.products_id, p.products_sort_order, p.products_price, p.products_model
+                FROM " . TABLE_PRODUCTS . " p"
+                . $this->categories_join . "
+                INNER JOIN " . TABLE_PRODUCTS_DESCRIPTION . " pd ON (p.products_id = pd.products_id) 
+				" . $this->attributes_join . "
+				WHERE pd.language_id = " . (int)$_SESSION['languages_id'];
+        }
+
+
+        protected function processSQL()
+        {
+            global $currencies;
+
+            $this->setSQL();
+            $this->runSQL();
+
+            $parm_2 = '';
+            $parm_3 = '';
+
+            if ($this->show_model) {
+                $parm_2 = '%2$s';
+            }
+
+            if ($this->show_price) {
+                $parm_3 = ' (%3$s)';
+            }
+
+            $this->output_string = '%1$s ' . $parm_2 . $parm_3; // format string with name first
+
+            if (strpos($this->sort,
+                    'model') && (!empty($this->attributes_join))) { // show model first only when there are attributes
```

Comment:
eeks, the formatting here seems odd with a line-break mid-function

Reply:
the nuances of settings in my IDE!  i hope to get it under control before the end of the decade...